fcrepo-clients/src/main/java/edu/unc/lib/dl/acl/fcrepo4/InheritedPermissionEvaluator.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2017 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.acl.fcrepo4;
+
+import static edu.unc.lib.dl.acl.util.PrincipalClassifier.classifyPrincipals;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import edu.unc.lib.dl.acl.util.Permission;
+import edu.unc.lib.dl.fedora.ContentPathFactory;
+import edu.unc.lib.dl.fedora.PID;
+
+/**
+ * Evaluates the permissions of agents on specific objects, taking into account
+ * inherited access control information
+ * 
+ * @author bbpennel
+ *
+ */
+public class InheritedPermissionEvaluator {
```

Comment:
I'm wondering whether it might make sense to break this class into two or three classes: one each for evaluating staff perms and patron perms, and a utility class with the responsibility for classifying principals. It might simplify some of the logic, as well as make the code more maintainable if we ever want to tweak patron or staff access without touching the other. Thoughts?